reference-implementation/lib/writable-stream.js

```diff
@@ -163,13 +163,37 @@ function WritableStreamAbort(stream, reason) {
   return promise;
 }
 
-function WritableStreamFinishAbort(stream) {
-  const error = new TypeError('Aborted');
-
+function WritableStreamError(stream, error) {
   stream._state = 'errored';
   stream._storedError = error;
 
-  WritableStreamRejectPromisesInReactionToError(stream);
+  // TODO: layering violation?
+  ResetQueue(stream._writableStreamController);
```

Comment:
It will probably lead to some extra churn when we add a ByteController, but as you say, we already have this issue in other places and in this case it's probably an easy fix (call stream.[[writableStreamController]].[[ResetQueue]] virtual method or similar).